test/server/cards/01_SOR/events/OverwhelmingBarrage.spec.js

```diff
@@ -0,0 +1,102 @@
+                expect(this.player1).toBeAbleToSelectExactly([this.battlefieldMarine, this.jynErso, this.atst, this.tielnFighter, this.bobaFett]);
+                expect(this.player1).toHaveChooseNoTargetButton();
+                this.player1.setDistributeDamagePromptState(new Map([
+                    [this.atst, 2],
+                    [this.battlefieldMarine, 2],
+                    [this.tielnFighter, 1],
+                    [this.bobaFett, 1]
+                ]));
+
+                expect(this.jynErso.damage).toBe(0);
+                expect(this.wampa.damage).toBe(0);
+                expect(this.atst.damage).toBe(2);
+                expect(this.battlefieldMarine.damage).toBe(2);
+                expect(this.tielnFighter).toBeInLocation('discard');
+                expect(this.bobaFett.damage).toBe(1);
+
+                // attack into wampa to confirm stats buff
+                this.atst.damage = 0;
+                this.player2.clickCard(this.atst);
+                this.player2.clickCard(this.wampa);
+                expect(this.wampa).toBeInLocation('ground arena');
+                expect(this.wampa.damage).toBe(6);
+                expect(this.atst).toBeInLocation('ground arena');
+                expect(this.atst.damage).toBe(6);
+            });
+
+            it('should be able to put all damage on a single target and exceed its HP total', function () {
+                this.player1.clickCard(this.overwhelmingBarrage);
+                this.player1.clickCard(this.wampa);
+                expect(this.player1).toBeAbleToSelectExactly([this.battlefieldMarine, this.jynErso, this.atst, this.tielnFighter, this.bobaFett]);
+                this.player1.setDistributeDamagePromptState(new Map([
+                    [this.tielnFighter, 6]
+                ]));
+
+                expect(this.jynErso.damage).toBe(0);
+                expect(this.wampa.damage).toBe(0);
+                expect(this.atst.damage).toBe(0);
+                expect(this.wampa.damage).toBe(0);
+                expect(this.tielnFighter).toBeInLocation('discard');
+                expect(this.bobaFett.damage).toBe(0);
+            });
+
+            it('should be able to choose 0 targets', function () {
+                this.player1.clickCard(this.overwhelmingBarrage);
+                this.player1.clickCard(this.wampa);
+                expect(this.player1).toBeAbleToSelectExactly([this.battlefieldMarine, this.jynErso, this.atst, this.tielnFighter, this.bobaFett]);
+                this.player1.clickPrompt('Choose no targets');
+
+                expect(this.jynErso.damage).toBe(0);
+                expect(this.wampa.damage).toBe(0);
+                expect(this.atst.damage).toBe(0);
+                expect(this.wampa.damage).toBe(0);
+                expect(this.tielnFighter.damage).toBe(0);
+                expect(this.bobaFett.damage).toBe(0);
+                expect(this.player2).toBeActivePlayer();
+            });
```

Comment:
Should probably make sure the player can't assign less than the full damage in total(comp rules 8.5.5).

Reply:
Right now I'm expecting that it will be the responsibility of the FE to enforce that the amount falls in the legal range. I do have checks in the prompt that confirm the amount is in the valid range, they throw an exception if so.